### PR TITLE
Ticket Sale: credit card checkout

### PR DESCRIPTION
**Pull request by schulzalexander**
The credit card checkout is not finished yet, but in my opinion ready for the demo.

Things that are still left to do:

- find out how to retrieve user's ID (the one that is used as primary key in the database), which is needed to reserve a ticket (by now, one will always "purchase" the same ticket with a static ID)
- when the user presses the purchase button, the TUMCabeClient returns a static ticket, because the recently merged purchasing-backend-logic is not compatible with the float-value price of the current database content, we have to wait here until the new sample data has been added to the database to use the actual purchasing server endpoint.

### CODE REVIEW

**Comment by schulzalexander** [comment]
@Kailiangdong yes the crash is caused by the second bullet point I stated above, the target Activity couldnt retrieve the ticket from the Ticket database because it wasn't added.
I fixed this, so now when the user presses purchase, a static ticket is added to the database and the Intent works.
In order to use the purchasing endpoint that returns a real ticket, the new sample data has to be added to be backend's databases first.

**Comment by Kailiangdong** [comment]
@schulzalexander  it works .looks good. It seems I don't have right to  approve it anymore. right?

File: app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
 import de.tum.in.tumcampusapp.utils.sync.SyncDao;
 import de.tum.in.tumcampusapp.utils.sync.model.Sync;
 
-@Database(version = 10, entities = {
+@Database(version = 11, entities = {
```
Comment (saileru) [inline app/src/main/java/de/tum/in/tumcampusapp/database/TcaDb.java]: Thanks ;)

**Comment by saileru** [comment]
I get the following exception when I click "Continue to payment"

**Comment by saileru** [comment]
I get the following exception when I click on "Continue to Payment"
 
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: de.tum.in.tumcampus, PID: 2396
                  java.lang.RuntimeException: Unable to destroy activity {de.tum.in.tumcampus/de.tum.in.tumcampusapp.component.ui.ticket.StripePaymentActivity}: android.os.NetworkOnMainThreadException
                      at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4347)
                      at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4365)
                      at android.app.ActivityThread.-wrap5(ActivityThread.java)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1571)
                      at android.os.Handler.dispatchMessage(Handler.java:111)
                      at android.os.Looper.loop(Looper.java:227)
                      at android.app.ActivityThread.main(ActivityThread.java:6102)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:961)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:822)
                   Caused by: android.os.NetworkOnMainThreadException
                      at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1288)
                      at java.net.InetAddress.lookupHostByName(InetAddress.java:432)
                      at java.net.InetAddress.getAllByNameImpl(InetAddress.java:253)
                      at java.net.InetAddress.getAllByName(InetAddress.java:215)
                      at okhttp3.Dns$1.lookup(Dns.java:40)
                      at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:185)
                      at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.java:149)
                      at okhttp3.internal.connection.RouteSelector.next(RouteSelector.java:84)
                      at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:214)
                      at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
                      at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
                      at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
                      at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
                      at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
                      at de.tum.in.tumcampusapp.api.app.ConnectivityInterceptor.intercept(ConnectivityInterceptor.kt:20)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
                      at de.tum.in.tumcampusapp.api.app.Helper.lambda$getDeviceInterceptor$1$Helper(Helper.java:106)
                      at de.tum.in.tumcampusapp.api.app.Helper$$Lambda$1.intercept(Unknown Source)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                      at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
                      at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
                      at okhttp3.RealCall.execute(RealCall.java:77)
                      at retrofit2.OkHttpCall.execute(OkHttpCall.java:180)
                      at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall.execute(ExecutorCallAdapterFactory.java:91)
                      at de.tum.in.tumcampusapp.api.app.TUMCabeClient.cancelTicketReservation(TUMCabeClient.java:386)
                      at de.tum.in.tumcampusapp.component.ui.ticket.StripePaymentActivity.onDestroy(StripePaymentActivity.java:88)
                      at android.app.Activity.performDestroy(Activity.java:6450)
                      at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1167)
                  	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:43

**Comment by schulzalexander** [comment]
@saileru I removed an unnecessary function call, does it work now for you?

**Pull request merged**